pkg/sql/tenant.go

```diff
@@ -261,7 +261,11 @@ func updateTenantRecord(
 func (p *planner) CreateTenant(
 	ctx context.Context, tenantName roachpb.TenantName,
 ) (roachpb.TenantID, error) {
-	if err := p.RequireAdminRole(ctx, "create tenant"); err != nil {
+	const op = "create tenant"
+	if err := p.RequireAdminRole(ctx, op); err != nil {
+		return roachpb.TenantID{}, err
+	}
+	if err := rejectIfCantCoordinateMultiTenancy(p.ExecCfg().Codec, op); err != nil {
```

Comment:
what does this change do?

Reply:
`rejectIfCantCoordinateMultiTenancy` ensures that only the system tenant can call `CreateTenant`. Previously, `CreateTenant` would fail later when it goes to write to `system.tenants` since that table doesn't exist in a secondary tenant. Instead, we get a better error by calling this method at the top of `CreateTenant`.